packages/core/src/agent/message-list/index.ts

```diff
@@ -1497,6 +1497,13 @@ export class MessageList {
             break;
 
           case 'tool-call':
+            // Merge part-level and message-level providerOptions
+            // Part-level takes precedence over message-level
+            const toolCallProviderMetadata = {
+              ...('providerOptions' in coreMessage && coreMessage.providerOptions ? coreMessage.providerOptions : {}),
+              ...('providerOptions' in part && part.providerOptions ? part.providerOptions : {}),
+            };
```

Comment:
@TylerBarnes I think you had some thoughts on adding the providerOptions here. Maybe we only want to include the providerOptions on the individual parts instead of also adding from the top level message?